std/datetime/date.d

```diff
@@ -6594,12 +6594,12 @@ public:
             $(REF DateTimeException,std,datetime,date) if the given day is an
             invalid day of the year.
       +/
-    @property void dayOfYear(int day) @safe pure
+    @property void dayOfYear(int day) @safe pure nothrow @nogc
     {
         immutable int[] lastDay = isLeapYear ? lastDayLeap : lastDayNonLeap;
 
         if (day <= 0 || day > (isLeapYear ? daysInLeapYear : daysInYear))
-            throw new DateTimeException("Invalid day of the year.");
+            assert(0, "Invalid day of the year.");
```

Comment:
@ZombineDev Andrei has been against such breaking changes in the past and the solution that @wilzbach proposed was originally put forward by him.

The problem is that the singleton approach has its own issues and is not really a solution.

>Does anyone know the exact status of DIP1008?

AFAIK Walter said it was on hold